dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -158,11 +167,11 @@ public void onFailure(Throwable throwable) {
 
         SortableChannel computeScore() {
             int requestsInflight = inflight.get();
-            int recentFailures = recentFailuresReservoir.get();
+            double failureReservoir = recentFailuresReservoir.get();
 
-            int score = requestsInflight + recentFailures;
+            int score = requestsInflight + Ints.saturatedCast(Math.round(failureReservoir));
```

Comment:
nit: might want to add a comment describing the case where a small fraction impacts selection long beyond its intended lifespan in the absence of other data.

Reply:
Even better, have captured it in a test.

Reply:
Tests are great, but it's also nice to provide inline context at a glance ;-)